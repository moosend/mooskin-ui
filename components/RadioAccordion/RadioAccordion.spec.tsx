import toJson from 'enzyme-to-json';
import * as React from 'react';
import Accordion, {Header, RadioAccordionContent} from './RadioAccordion';

import { mount, render, shallow } from 'enzyme';

describe('RadioAccordion', () => {

    test('renders RadioAccordion properly according to snapshot', () => {

        Date.now = jest.fn(() => 1482363367071);
        Math.random = jest.fn(() => 222333444555);

        const component = shallow(
            <Accordion id="5" className="mySwitch" style={{color: 'black'}}>
                <RadioAccordionContent title="title1">Content1</RadioAccordionContent>
                <RadioAccordionContent title="title2" active>Content2</RadioAccordionContent>
                <RadioAccordionContent title="title3" className="classname">Content3</RadioAccordionContent>
            </Accordion>
        );

        expect(component.find('Header').length).toBe(3);
        expect(component.find(RadioAccordionContent).length).toBe(3);

        expect(component.find(RadioAccordionContent).first().prop('children')).toBe('Content1');

        expect(toJson(component)).toMatchSnapshot();
    });

    test('renders Content properly according to snapshot', () => {

        Math.random = jest.fn(() => 222333444555);

        const component = shallow(
            <RadioAccordionContent
                title="title1"
                style={{color: 'blue'}}
                active
            >
                Some Content
            </RadioAccordionContent>
        );

        expect(toJson(component)).toMatchSnapshot();
    });

    test('renders Header properly according to snapshot', () => {

        Date.now = jest.fn(() => 1482363367071);

        const component = shallow(
            <Header
                title="title1"
                onClick={() => null}
                active
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });

    test('changes the state and displays appropriate content when one of the headers is clicked and back', () => {

        const component = mount(
            <Accordion>
                <RadioAccordionContent title="title1">Content1</RadioAccordionContent>
                <RadioAccordionContent title="title2" active>Content2</RadioAccordionContent>
            </Accordion>
        );

        expect(component.state('active')).toBe(1);
        expect(component.find('.accordion-header').first().find('input').prop('checked')).toBeFalsy();
        expect(component.find('.accordion-header').last().find('input').prop('checked')).toBeTruthy();
        expect(component.find('.accordion-content').first().hasClass('invisible')).toBeTruthy();
        expect(component.find('.accordion-content').last().hasClass('visible')).toBeTruthy();

        component.find('.radio').first().simulate('click');

        expect(component.state('active')).toBe(0);
        expect(component.find('.accordion-header').first().find('input').prop('checked')).toBeTruthy();
        expect(component.find('.accordion-header').last().find('input').prop('checked')).toBeFalsy();
        expect(component.find('.accordion-content').first().hasClass('visible')).toBeTruthy();
        expect(component.find('.accordion-content').last().hasClass('invisible')).toBeTruthy();

        component.find('.radio').last().simulate('click');

        expect(component.state('active')).toBe(1);
        expect(component.find('.accordion-header').first().find('input').prop('checked')).toBeFalsy();
        expect(component.find('.accordion-header').last().find('input').prop('checked')).toBeTruthy();
        expect(component.find('.accordion-content').first().hasClass('invisible')).toBeTruthy();
        expect(component.find('.accordion-content').last().hasClass('visible')).toBeTruthy();

    });
});
