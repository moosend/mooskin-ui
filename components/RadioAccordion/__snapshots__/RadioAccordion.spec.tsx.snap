// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioAccordion renders Content properly according to snapshot 1`] = `
<div
  className="accordion-content content visible"
  style={
    Object {
      "color": "blue",
    }
  }
>
  Some Content
</div>
`;

exports[`RadioAccordion renders Header properly according to snapshot 1`] = `
<div
  className="accordion-header container"
>
  <div
    className="radio activeHeader checkedRadio"
    onClick={[Function]}
  >
    <input
      checked={true}
      id="4zlbrf"
      readOnly={true}
      type="radio"
    />
    <label
      htmlFor="4zlbrf"
    >
      <span>
        title1
      </span>
    </label>
  </div>
  <div />
</div>
`;

exports[`RadioAccordion renders RadioAccordion properly according to snapshot 1`] = `
<div
  className="accordion-component container"
>
  <Header
    active={false}
    key="0"
    name="1482363367071"
    onClick={[Function]}
    title="title1"
  >
    <RadioAccordionContent
      active={false}
      key="0"
      title="title1"
    >
      Content1
    </RadioAccordionContent>
  </Header>
  <Header
    active={true}
    key="1"
    name="1482363367071"
    onClick={[Function]}
    title="title2"
  >
    <RadioAccordionContent
      active={true}
      key="1"
      title="title2"
    >
      Content2
    </RadioAccordionContent>
  </Header>
  <Header
    active={false}
    key="2"
    name="1482363367071"
    onClick={[Function]}
    title="title3"
  >
    <RadioAccordionContent
      active={false}
      className="classname"
      key="2"
      title="title3"
    >
      Content3
    </RadioAccordionContent>
  </Header>
</div>
`;
